prov/efa/src/rdm/efa_rdm_ep.h

```diff
@@ -508,14 +508,26 @@ static inline int efa_rdm_attempt_to_sync_memops(struct efa_rdm_ep *ep, void *bu
 static inline int efa_rdm_attempt_to_sync_memops_iov(struct efa_rdm_ep *ep, struct iovec *iov, void **desc, int num_desc)
 {
 	int err = 0, i;
+	struct efa_mr *efa_mr;
 
 	if (!desc)
 		return err;
 
-	for (i = 0; i < num_desc; i++) {
-		err = efa_rdm_attempt_to_sync_memops(ep, iov[i].iov_base, (struct efa_mr *) desc[i]);
-		if (err)
-			return err;
+	if (OFI_UNLIKELY(ep->cuda_api_permitted)) {
```

Comment:
Can we move this cuda_api_permitted check out of `efa_rdm_attempt_to_sync_memops`, so `efa_rdm_attempt_to_sync_memops_iov` and `efa_rdm_attempt_to_sync_memops_ioc` can still reuse `efa_rdm_attempt_to_sync_memops`

Reply:
You can do something like

```
if (ep->cuda_api_permitted && efa_rdm_attempt_to_sync_memops_iov(efa_rdm_ep, (struct iovec *)msg->msg_iov, msg->desc, msg->iov_count))
		return ret;
```

Reply:
I removed `efa_rdm_attempt_to_sync_memops` instead. We setup the IOVs anyway in the TX path.